ResearchKit/Common/ORKSurveyAnswerCellForText.m

```diff
@@ -244,7 +245,18 @@ - (void)prepareView {
 }
 
 - (BOOL)shouldContinue {
-    return ![self correctValueIfNeeded];
+    ORKTextAnswerFormat *answerFormat = (ORKTextAnswerFormat *)[self.step impliedAnswerFormat];
+    if (answerFormat.isEmail) {
+        BOOL isValid = [answerFormat isAnswerValidWithString:self.answer];
+
+        if (!isValid){
+            [self showValidityAlertWithMessage:self.answer];
```

Comment:
Should implement `localizedInvalidValueStringWithAnswerString:` to give user a better alert message.
```
 [self showValidityAlertWithMessage:[[self.step impliedAnswerFormat] localizedInvalidValueStringWithAnswerString:self.textField.text]];
```